sdk/trace/span.go

```diff
@@ -163,6 +163,35 @@ func (s *span) Event(ctx context.Context, msg string, attrs ...core.KeyValue) {
 	s.mu.Unlock()
 }
 
+func (s *span) SetName(name string) {
+	if s.data == nil {
+		// TODO: now what?
+		return
+	}
+	s.data.Name = name
+	// SAMPLING
+	noParent := s.data.ParentSpanID == 0
+	sd := samplingDecision{
+		noParent:     noParent,
+		remoteParent: s.data.HasRemoteParent,
+	}
+	sd.makeDecision(func() samplingData {
+		return samplingData{
+			parent: func() core.SpanContext {
```

Comment:
I wonder if these `func()` here and above are intended to help readability? They feel not very Go-like to me. I would probably write this section of code as:

```
	var sctx core.SpanContext
	if noParent {
	    sctx = core.EmptySpanContext()
	} else {
	    ctx = s.data.SpanContext
	}
	sd.makeDecision(samplingData{
			parent: sctx,
			name: name,
			cfg:  config.Load().(*Config),
			span: s,
	})
```

Reply:
That's what I usually do when I have some kind of a conditional initialization of the variable. I suppose that this feel a bit like C++. ;)

Will change it.